drivers/can/can_sam0.c

```diff
@@ -113,20 +113,22 @@ static int can_sam0_get_core_clock(const struct device *dev, uint32_t *rate)
 
 static void can_sam0_clock_enable(const struct can_sam0_config *cfg)
 {
+	uint8_t gen_index = cfg->gclk_gen;
+
 	*cfg->mclk |= cfg->mclk_mask;
 
 	GCLK->PCHCTRL[cfg->gclk_id].reg = GCLK_PCHCTRL_CHEN
-					| GCLK_PCHCTRL_GEN(cfg->gclk_gen);
+					| GCLK_PCHCTRL_GEN(gen_index);
```

Comment:
Hi,

What I think this driver does different for all others if that it configure a generic clock itself. IMHO this is SoC init responsibility until a proper driver be available.

Why:

This driver waists a generic line, see below:

DFLL clock is already on genclk[0] for SAME 5x and it is the default on devicetree, see https://github.com/zephyrproject-rtos/zephyr/blob/553fc84e77e245eda41dd1e2f2ba6ed1055d4bcb/soc/atmel/sam0/common/soc_samd5x.c#L168

Same for SAMC2x, see https://github.com/zephyrproject-rtos/zephyr/blob/553fc84e77e245eda41dd1e2f2ba6ed1055d4bcb/soc/atmel/sam0/common/soc_samc2x.c#L49

My suggestion is, looking in the original code: keep 116~119 and drop all lines between 120~132. Revert the `atmel,assigned-clocks` to use the default.